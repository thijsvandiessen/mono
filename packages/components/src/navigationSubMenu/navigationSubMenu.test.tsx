import { describe, expect, it } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import { NavigationSubMenu } from '../navigationSubMenu/index.js'
import React from 'react'

const mockData = [
  {
    id: '194164582',
    label: 'specific page',
    link: { slug: 'specific/page' },
  },
  {
    id: '194164583',
    label: 'homepage',
    link: { slug: 'homepage', title: 'home' },
  },
  {
    id: '194164584',
    label: 'external',
    link: { slug: 'https://example.com' },
  },
]

describe('navigationSubMenu', () => {
  it('should render a button', () => {
    const { container } = render(
      <NavigationSubMenu label="root label" item={JSON.stringify(mockData)} />
    )

    expect(screen.getByRole('button', { name: 'root label' })).toBeVisible()
    expect(container).toMatchSnapshot()

    fireEvent(
      screen.getByRole('button'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    )

    expect(screen.getByRole('link', { name: 'specific page' })).toHaveAttribute(
      'href',
      '/specific/page'
    )
    expect(container).toMatchSnapshot()
  })
})
