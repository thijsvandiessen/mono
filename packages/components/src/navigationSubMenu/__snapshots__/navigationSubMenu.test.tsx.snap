// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`navigationSubMenu > should render a button 1`] = `
<div>
  <li
    class="_listItem_906789"
  >
    <button
      class="_button_906789 text-large"
      type="button"
    >
      root label
    </button>
  </li>
</div>
`;

exports[`navigationSubMenu > should render a button 2`] = `
<div>
  <li
    class="_listItem_906789"
  >
    <button
      class="_button_906789 text-large"
      type="button"
    >
      root label
    </button>
    <ul
      class="_list_906789"
    >
      <li
        class="_navItem_16c685"
      >
        <a
          class="_link_16c685 text-large"
          href="/specific/page"
        >
          specific page
        </a>
      </li>
      <li
        class="_navItem_16c685"
      >
        <a
          class="_link_16c685 text-large"
          href="/"
        >
          homepage
        </a>
      </li>
      <li
        class="_navItem_16c685"
      >
        <a
          class="_link_16c685 text-large"
          href="https://example.com"
        >
          external
        </a>
      </li>
    </ul>
    <div
      class="_arrow_906789"
    />
  </li>
</div>
`;
